json.go

```diff
@@ -55,7 +56,12 @@ func (om *OrderedMap[K, V]) MarshalJSON() ([]byte, error) {
 		case uint64:
 			writer.Uint64Str(key)
 		default:
-			return nil, fmt.Errorf("unsupported key type: %T", key)
+			keyRefl := reflect.ValueOf(key)
+			if keyRefl.Type().Kind() == reflect.String {
```

Comment:
Great idea. Would actually be even better to do the same for all int types that we support.

Reply:
Done in https://github.com/wk8/go-ordered-map/pull/23